docs/design/ovs-pipeline.md

```diff
@@ -1,1197 +1,1852 @@
+for remote Endpoints, like the bucket with `bucket_id` 1 in this group.
 
-If the `conjunction` action is matched, packets are "allowed" or "dropped"
-based on the `action` field of the policy rule. If allowed, they follow a similar
-path as described in the following [EgressRuleTable] section.
+The third group with `group_id` 11 is the destination of packets matched by flow 6, designed for a Service that has
+Endpoints and is configured with session affinity. The group closely resembles the group with `group_id` 10, except that
+the destination of the packets is table [ServiceLB], rather than table [EndpointDNAT]. After being sent back to table
+[ServiceLB], they will be matched by flow 7.
 
-Unlike the default of K8s NetworkPolicies, Antrea-native policies have no such
-default rules. Hence, they are evaluated as-is, and there is no need for a
-AntreaPolicyEgressDefaultTable.
+### EndpointDNAT
 
-### EgressRuleTable (50)
+The table implements DNAT for Service connection after Endpoint selection is performed in table [ServiceLB].
 
-For this table, you will need to keep mind the Network Policy
-[specification](#network-policy-implementation) that we are using. We have 2
-Pods running on the same Node, with IP addresses 10.10.1.2 to 10.10.1.3. They
-are allowed to talk to each other using TCP on port 80, but nothing else.
-
-This table is used to implement the egress rules across all Network Policies. If
-you dump the flows for this table, you should see something like this:
+If you dump the flows of this table, you may see the following::
 
 ```text
-1. table=50, priority=210,ct_state=-new+est,ip actions=goto_table:70
-2. table=50, priority=200,ip,nw_src=10.10.1.2 actions=conjunction(2,1/3)
-3. table=50, priority=200,ip,nw_src=10.10.1.3 actions=conjunction(2,1/3)
-4. table=50, priority=200,ip,nw_dst=10.10.1.2 actions=conjunction(2,2/3)
-5. table=50, priority=200,ip,nw_dst=10.10.1.3 actions=conjunction(2,2/3)
-6. table=50, priority=200,tcp,tp_dst=80 actions=conjunction(2,3/3)
-7. table=50, priority=190,conj_id=2,ip actions=load:0x2->NXM_NX_REG5[],ct(commit,table=61,zone=65520,exec(load:0x2->NXM_NX_CT_LABEL[32..63]))
-8. table=50, priority=0 actions=goto_table:60
+1. table=EndpointDNAT, priority=200,reg0=0x4000/0x4000 actions=controller(reason=no_match,id=62373,userdata=04)
+2. table=EndpointDNAT, priority=200,tcp,reg3=0xa0a0018,reg4=0x20050/0x7ffff actions=ct(commit,table=AntreaPolicyEgressRule,zone=65520,nat(dst=10.10.0.24:80),exec(set_field:0x10/0x10->ct_mark,move:NXM_NX_REG0[0..3]->NXM_NX_CT_MARK[0..3]))
+3. table=EndpointDNAT, priority=200,tcp,reg3=0xa0a0106,reg4=0x20050/0x7ffff actions=ct(commit,table=AntreaPolicyEgressRule,zone=65520,nat(dst=10.10.1.6:80),exec(set_field:0x10/0x10->ct_mark,move:NXM_NX_REG0[0..3]->NXM_NX_CT_MARK[0..3]))
+4. table=EndpointDNAT, priority=190,reg4=0x20000/0x70000 actions=set_field:0x10000/0x70000->reg4,resubmit(,ServiceLB)
```

Comment:
I noticed that you didn't describe this flow below. Can you add a quick description?

Reply:
Will add. I forgot the description of flows 4 and 5 in this table.